### DISCUSSION TITLE
What does the $ in the model definition xacro mean

### DISCUSSION THREAD

**Discussion by throwawayacco** [ctu-mrs/mrs_uav_system#70] [Q&A]
So in any of the model xacros, example f450.xacro all the intertia definitions are has a $ infront of them. The problem is that when using a precalculated value there is a huge difference between using the $ or not

So 

Ixx = "0.1"
Iyy = "0.1"
Izz = "0.1"
and 
Ixx = "$0.1"
Iyy = "$0.1"
Izz = "$0.1"
behave in completly different ways when simulated. 

What does the $ sign mean and which definition is correct?

**Comment by throwawayacco** [comment]
I'm refering specifically to line 118,119 and 120. 

An additional question, how has the inertia body radius and inertia body height in line 76&77 been determined/calculated?

**Comment by spurnvoj** [comment]
Hi, check the documentation for xacro
http://wiki.ros.org/xacro

ne 9. 5. 2021 v 3:56 odesílatel throwawayacco ***@***.***>
napsal:

**Comment by spurnvoj** [comment]
The Inertia of the body of the drone is simplified and it is represented using a cylinder.. how the inertia of the cylinder is computed is explained for example here https://scienceworld.wolfram.com/physics/MomentofInertiaCylinder.html. The
parameters of the cylinder are specified in the same file (see https://github.com/ctu-mrs/mrs_simulation/blob/master/models/mrs_robots_description/urdf/f450.xacro#L76-L77)
